int32 desiredState
---
bool executed